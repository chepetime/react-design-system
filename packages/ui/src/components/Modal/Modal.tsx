import React from "react";
import classnames from "classnames";
import css from "./Modal.module.scss";

export interface ModalProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const Modal = React.forwardRef<HTMLDivElement, ModalProps>(
  (props: ModalProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.Modal)} ref={ref}>
        <p>Modal component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

Modal.displayName = "Modal";
