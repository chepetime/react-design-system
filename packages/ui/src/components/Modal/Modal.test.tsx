
import React from "react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { Modal } from "./Modal";
import { render, screen } from "@testing-library/react";

describe("Modal", () => {
  it("renders without errors", () => {
    render(<Modal />);
    // Assert that the component renders without throwing any errors
  });

  it("renders with children", () => {
    render(<Modal>Some children</Modal>);
    // Assert that the component renders the provided children
    const childrenElement = screen.getByText("Some children");
    expect(childrenElement).toBeInTheDocument();
  });

  it("handles user interactions", () => {
    render(<Modal>Some children</Modal>);
    // Simulate user interactions with the component
    userEvent.click(screen.getByText("Some children"));
    // Assert the expected outcome of the user interaction
    // For example, check if a certain state has changed or an event handler has been called
  });

  it("matches snapshot", () => {
    const { container } = render(<Modal />);
    // Assert that the rendered component matches a previously captured snapshot
    expect(container).toMatchSnapshot();
  });
});
