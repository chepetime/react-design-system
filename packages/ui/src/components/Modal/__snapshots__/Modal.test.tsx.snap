// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal matches snapshot 1`] = `
<div>
  <div
    class="Modal"
  >
    <p>
      Modal component working!
    </p>
    <div />
  </div>
</div>
`;
